src/crypto/crypto.h

```diff
@@ -278,12 +292,22 @@ namespace crypto {
   inline std::ostream &operator <<(std::ostream &o, const crypto::signature &v) {
     epee::to_hex::formatted(o, epee::as_byte_span(v)); return o;
   }
+  inline std::ostream &operator <<(std::ostream &o, const crypto::legacy16B_secret_key &v) {
+    epee::to_hex::formatted(o, epee::as_byte_span(v)); return o;
+  }
 
   const static crypto::public_key null_pkey = boost::value_initialized<crypto::public_key>();
   const static crypto::secret_key null_skey = boost::value_initialized<crypto::secret_key>();
+  const static crypto::legacy16B_secret_key null_legacy16B_skey = boost::value_initialized<crypto::legacy16B_secret_key>();
+
+  static unsigned long sec_seed_bytes_length = 32;
+  static unsigned long legacy16B__sec_seed_bytes_length = 16;
+  static unsigned long sec_seed_hex_string_length = sec_seed_bytes_length * 2;
+  static unsigned long legacy16B__sec_seed_hex_string_length = legacy16B__sec_seed_bytes_length * 2;
```

Comment:
Probably, thanks. My C++ inexperience shows. :)

Reply:
Are these constants really needed in the first place?

Reply:
They were the way that I decided to encapsulate the declaration of constants I use for detecting seed and mnemonic size. electrum-words has something similar. I am very very open to suggestions as I am inexperienced in C++ and haven't developed my best practices yet.

Reply:
The general principle in my understanding is that every new code must be introduced because it's necessary and used. Unused additional code should be avoided.

Reply:
Code is used, If I may ask you to refer to my other comment.

Reply:
Can you provide a link?

Reply:
https://github.com/monero-project/monero/pull/3281#discussion_r170424422

and, may I also suggest it as a best practice for simplewallet to use the electrum words seed word count constant declaration instead?

Reply:
These constants have been unnecessary in the codebase because we can just use `sizeof(secret_key)` and the hex string length is just the twice of the byte length. I still don't see the need for them.

Reply:
Re `sizeof(type)` very good point. Thanks. That lets me remove two of the four. 

So, to explain myself and my view of this case a bit, I do see the usefulness of encapsulating the hex string length calculation behind an 'accessor' style variable. It's amusingly virtually the most minimal case where such an argument would be valid imo but it makes sense in this case because people who want to do such detection would need such values and therefore the calculation would be spread across multiple places. The main argument I see for not having to add such declarations to a central relevant domain such as crypto is that we pretty much expect the derivation to (a) remain fairly trivial and (b) basically always have the same coefficient of `2`. But the code is accurate, relevant, and imo legitimately useful to library consumers, and so the implementation would end up getting duplicated across the codebase (as you can see my provided example code does use it with a slight readability advantage). But I think that this case is so trivial that it doesn't matter either way, so feel free to let me know how you'd like to close this one out. :) d2dc91f0 for your review